### DISCUSSION TITLE
How to assign a function object to property of the napi object?

### DISCUSSION THREAD

**Discussion by yuchanns** [napi-rs/napi-rs#1173] [Q&A] [answered]
A javascript object can be assigned with a function object as its property like:
```js
let extendableObj = { name: "example", extendsPlugin: (p) => { p.use(somePlugin) } }
```
How to do when it comes to napi object?

Try:
```rust
#[napi(object)]
pub struct ExtendableObj {
    pub name: &'static str,
    pub extends_plugin: Box<dyn Fn()>,
}
```
Not work as `the trait bound Box<dyn Fn()>: NapiRaw is not satisfied`, neither to `fn()`.

**Comment by Brooooooklyn** [comment, marked as answer]
`Box<dyn Fn>` is not supported to create from JavaScript function, use `JsFunction` instead
